SECURITY.md

```diff
@@ -0,0 +1,15 @@
+# Security Policy
+
+This document outlines the security policy for the Datadog Ruby client library (aka Ruby tracer) and what to do if you discover a security vulnerability in the project.
+Most notably, please do not share the details in a public forum (such as in a discussion, issue, or pull request) but instead reach out to us with the details.
+This gives us an opportunity to release a fix for others to benefit from by the time details are made public.
+
+## Supported Versions
+
+We accept vulnerability submissions for the [currently maintained release](https://github.com/DataDog/dd-trace-rb/releases).
```

Comment:
I don't think it is clear what "the currently maintained release is". My understanding is there are at least two since we do maintain both 1.x and 2.x branches. But we don't indicate which releases are maintained on the releases page - it just lists available releases.

I am not sure what the correct wording should be here but I think it would be helpful to provide at least a link to where one can figure out whether a particular release is maintained or not.

Reply:
I can make "release" plural:
```md
We accept vulnerability submissions for the 
[currently maintained releases](https://github.com/DataDog/dd-trace-rb/releases).
```

Essentially the visitor visits the releases page, sees that 1.x and 2.x is maintained, and knows not to file a bug report for an issue that they discover in a 0.x release.

All tracers have a releases page so that's the easiest way to ascertain what is current. The Node.js tracer goes further and explicitly maintains a tables of releases and their status but that might be too much to ask of other tracers:
https://github.com/DataDog/dd-trace-js?tab=readme-ov-file#version-release-lines-and-maintenance